doc/user/sql/create-source/avro-source.md

```diff
@@ -0,0 +1,188 @@
+menu:
+  main:
+    parent: 'create-source'
+aliases:
+    - /docs/sql/create-source/kafka
+    - /docs/sql/create-source/avro
+---
+
+`CREATE SOURCE` connects Materialize to some data source, and lets you interact
+with its data as if it were in a SQL table.
+
+This document details how to connect Materialize to an Avro-formatted Kafka
+topic. For other options, view [`CREATE  SOURCE`](../).
+
+## Conceptual framework
+
+Sources represent connections to resources outside Materialize that it can read
+data from. For more information, see [API Components:
+Sources](../../../overview/api-components#sources).
+
+## Syntax
+
+{{< diagram "create-source-avro.html" >}}
+
+Field | Use
+------|-----
+_src&lowbar;name_ | The name for the source, which is used as its table name within SQL.
+**KAFKA BROKER** _host_ | The Kafka broker's host name.
+**FORMAT ...** _url_ | The URL of the Confluent schema registry to get schema information from.
+**FORMAT ...** _schema&lowbar;file&lowbar;path_ | The absolute path to a file containing the schema.
+**FORMAT ...** _inline&lowbar;schema_ | A string representing the schema.
+**ENVELOPE** _envelope_ | The envelope type.<br/><br/> &#8226; **NONE** creates an append-only source. This means that records will only be appended and cannot be updated or deleted. <br/><br/>&#8226; **DEBEZIUM** creates a source that supports all CRUD-like operations. This option requires records have the [appropriate fields](#format-implications), and is generally only supported by sources published to Kafka by [Debezium].<br/><br/>For more information, see [Debezium envelope details](#debezium-envelope-details).
+
+## Details
+
+This document assumes you're using Kafka to send Avro-encoded data to
+Materialize.
+
+### Kafka source details
+
+Materialize expects each source to use to one Kafka topic, which is&mdash;in
+  turn&mdash;generated by a single table in an upstream database.
+
+### Avro format details
+
+Avro-formatted external sources require providing you providing the schema in
+one of two ways:
+
+- Using the [Confluent Schema Registry](#using-a-confluent-schema-registry)
+- Providing the Avro schema [in-line when creating the
+  source](#inlining-the-avro-schema).
+
+### Envelope details
+
+Envelopes determine whether an incoming record inserts new data, updates or
+deletes existing data, or both. For more information, see [API Components:
+Envelopes](../../../overview/api-components#envelopes).
+
+Avro-encoded data sent through Kafka is the only type of data that can support
+the Debezium envelope.
+
+### Debezium envelope details
+
+The Debezium envelope provides a "diff envelope", which describes the decoded
+records' old and new values; this is roughly equivalent to the notion of Change
+Data Capture, or CDC. Materialize can use the data in this diff envelope to
+process data as representing inserts, updates, or deletes.
+
+This envelope is called the Debezium envelope because the only tool that we're
```

Comment:
The format is actually very specific to Debezium -- this type of source wouldn't work (currently) even if there is another tool that provides exactly the same information, unless the field names it uses are exactly the same.

I think it's fine to say that this is called Debezium because it interprets records in Debezium's output format.

Reply:
Cockroach also uses the Debezium envelope format with exactly the same field names, FWIW.

Reply:
All the more reason not to say Debezium is "the only tool we're aware of" that does so ;)